### PR TITLE
Docs improvements

### PR DESCRIPTION
**Pull request by tcharding**
Do crate wide docs improvements.

### CODE REVIEW

File: src/hex.rs [hunk 1]
```diff
@@ -102,8 +102,11 @@ pub struct HexIterator<'a> {
 }
 
 impl<'a> HexIterator<'a> {
-    /// Constructs a new `HexIterator` from a string slice. If the string is of
-    /// odd length it returns an error.
+    /// Constructs a new `HexIterator` from a string slice.
+    ///
+    ///  # Errors
```
Comment (apoelstra) [inline src/hex.rs:107]: In a9ed7f8294c480ee42771d5b0a296b8f84806f3f:

There is an extra space before the `#`

Comment (tcharding) [reply src/hex.rs:107]: Cheers, I'd love to know how to lint the rustdocs :)

File: src/sha256.rs [hunk 2]
```diff
@@ -236,10 +236,11 @@ macro_rules! round(
...
 
 impl HashEngine {
-    /// Create a new [HashEngine] from a midstate.
+    /// Create a new [`HashEngine`] from a [`Midstate`].
     ///
-    /// Be aware that this method panics when [length] is
-    /// not a multiple of the block size.
+    /// # Panics
+    ///
+    /// Be aware that this method panics when `length` is not a multiple of the block size.
```
[the remark this answers is not in the document]
Comment (tcharding) [reply src/sha256.rs:243]: You got it, cheers.

**Comment by tcharding** [comment]
Since the two suggestions are technically improvements, and the docs build without them I, I did them as separate patches. Save you tabbing through all those boring docs changes again :)

**Review by apoelstra — APPROVED** [review]
ACK 734771b3c153e9ffbec8499b7b4c80aa0a36d55a

**Comment by apoelstra** [comment]
Needs rebase.

**Comment by tcharding** [comment]
Squashed the two above mentioned changes into the patch 2 since I had to rebase.

**Review by apoelstra — APPROVED** [review]
ACK ddfd8467c71ae76b8b5214e75713325d693dc9ce

**Pull request merged**